src/main/java/com/glodblock/github/client/gui/GuiPatternValueAmount.java

```diff
@@ -0,0 +1,244 @@
+            }
+
+            this.set.enabled = resultI > 0;
+        } catch (final NumberFormatException e) {
+            this.set.enabled = false;
+        }
+
+        this.amountToSet.drawTextBox();
+    }
+
+    @Override
+    protected void keyTyped(final char character, final int key) {
+        if (!this.checkHotbarKeys(key)) {
+            if (key == 28) {
+                this.actionPerformed(this.set);
+            }
+            this.amountToSet.textboxKeyTyped(character, key);
+            super.keyTyped(character, key);
+        }
+    }
+
+    @Override
+    protected void actionPerformed(final GuiButton btn) {
+        super.actionPerformed(btn);
+
+        try {
+
+            if (btn == this.originalGuiBtn) {
+                FluidCraft.proxy.netHandler.sendToServer(new CPacketSwitchGuis(this.originalGui));
+            }
+
+            if (btn == this.set && btn.enabled) {
+                double resultD = Calculator.conversion(this.amountToSet.getText());
+                int resultI;
+
+                if (resultD <= 0 || Double.isNaN(resultD)) {
+                    resultI = 1;
+                } else {
+                    resultI = (int) ArithHelper.round(resultD, 0);
+                }
+
+                FluidCraft.proxy.netHandler.sendToServer(new CPacketPatternValueSet(originalGui.ordinal(), resultI, valueIndex));
+            }
+        } catch (final NumberFormatException e) {
+            // nope..
+            this.amountToSet.setText("1");
+        }
+
+        final boolean isPlus = btn == this.plus1 || btn == this.plus10 || btn == this.plus100 || btn == this.plus1000;
+        final boolean isMinus = btn == this.minus1 || btn == this.minus10 || btn == this.minus100 || btn == this.minus1000;
+
+        if (isPlus || isMinus) {
+            this.addQty(this.getQty(btn));
+        }
+    }
+
+    private void addQty(final int i) {
+        try {
+            String out = this.amountToSet.getText();
+
+            double resultD = Calculator.conversion(out);
+            int resultI;
+
+            if (resultD <= 0 || Double.isNaN(resultD)) {
+                resultI = 0;
+            } else {
+                resultI = (int) ArithHelper.round(resultD, 0);
+            }
+
+            if (resultI == 1 && i > 1) {
+                resultI = 0;
+            }
+
+            resultI += i;
+            if (resultI < 1) {
+                resultI = 1;
+            }
+
+            out = Integer.toString(resultI);
+
+            this.amountToSet.setText(out);
+        } catch (final NumberFormatException e) {
+            // :P
+        }
+    }
+
+    private int getOriginalAmount(ItemStack stack) {
+        if (Util.isFluidPacket(stack)) {
+            FluidStack fluidStack = ItemFluidPacket.getFluidStack(stack);
+            return fluidStack.amount;
+        } else {
+            return stack.stackSize;
+        }
+    }
+
+    protected String getBackground() {
+        return "guis/craftAmt.png";
+    }
```

Comment:
why not use this method in L132

Reply:
I just copied the code from GuiFluidCraftAmount and modified it. In fact, this method is from ae, but ae doesn't use it either, and all GuiCraftAmount doesn't use it.

Reply:
then remove it